src/helpers.rs

```diff
@@ -790,3 +790,21 @@ pub fn simd_element_to_bool<'tcx>(elem: ImmTy<'tcx, Tag>) -> InterpResult<'tcx,
         _ => throw_ub_format!("each element of a SIMD mask must be all-0-bits or all-1-bits"),
     })
 }
+
+/// Retrieve the list of local crates that should have been passed by cargo-miri in
+/// MIRI_LOCAL_CRATES and turn them into `CrateNum`s
+pub fn get_local_crates(tcx: &TyCtxt<'_>) -> Vec<CrateNum> {
+    // Convert the local crate names from the passed-in config into CrateNums so that they can
+    // be looked up quickly during execution
+    let local_crate_names = std::env::var("MIRI_LOCAL_CRATES")
+        .map(|crates| crates.split(",").map(|krate| krate.to_string()).collect::<Vec<_>>())
+        .unwrap_or_default();
+    let mut local_crates = Vec::new();
+    for num in tcx.crates(()) {
+        let name = tcx.crate_name(*num).as_str().to_string();
+        if local_crate_names.contains(&name) {
```

Comment:
FWIW if you use `iter().any` you can avoid the `to_string` allocation in the previous line... not a big deal though.^^

Reply:
Do you mean this?
```diff
diff --git a/src/helpers.rs b/src/helpers.rs
index b4339cd8..563115b4 100644
--- a/src/helpers.rs
+++ b/src/helpers.rs
@@ -801,8 +801,8 @@ pub fn get_local_crates(tcx: &TyCtxt<'_>) -> Vec<CrateNum> {
         .unwrap_or_default();
     let mut local_crates = Vec::new();
     for num in tcx.crates(()) {
-        let name = tcx.crate_name(*num).as_str().to_string();
-        if local_crate_names.contains(&name) {
+        let name = tcx.crate_name(*num).as_str();
+        if local_crate_names.iter().any(|local_name| local_name == name) {
             local_crates.push(*num);
         }
     }
```
This doesn't compile because the `&str` I get doesn't live past the semicolon.

Reply:
Yeah, this should help:
```rust
let name = tcx.crate_name(*num);
let name = name.as_str();
```